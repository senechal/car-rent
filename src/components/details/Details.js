import {
  Link,
  useParams,
  Redirect,
} from 'react-router-dom';
import { DoubleLeftOutlined } from '@ant-design/icons';
import Vehicle from '@components/vehicle';
import { useSelector } from 'react-redux';
import  { Layout } from '@reusable';

const { Content } = Layout;

const Details = () => {
  const { vehId } = useParams();
  const vehicles = useSelector(state => state.vehicles);
  const vehicle = vehicles.find(veh => veh.id === vehId);


  return vehicles.length !== 0 && vehicle ? (
    <Content className="rac-details-content">
      <div className="rac-details-navigation">
        <Link to='/'> <DoubleLeftOutlined /> Go Back</Link>
      </div>
      <Vehicle {...vehicle}/>
    </Content>
  )
  : <Redirect to="/" />
};

export default Details;
