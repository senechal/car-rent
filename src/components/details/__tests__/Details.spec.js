import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import Details from '../Details';

const mockVehicle = {
  baggageQuantity: "2",
  currencyCode: "CAD",
  doors: "2",
  fuelType: "Petrol",
  hasAirConditioning: true,
  id: "HERTZ_1",
  imgUrl: "https://ctimg-fleet.cartrawler.com/kia/rio/primary.png?auto=format&w=600",
  makeModel: "Kia Rio or similar",
  price: "731.47",
  seats: "5",
  status: "Available",
  transmissionType: "Automatic",
  vendor: "HERTZ",
};

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  useParams: () => ({vehId: mockVehicle.id}),
  Redirect: () => '<Redirect to="/">',
  Link: ({children, to}) => <a href={to}>{children}</a>,
}));

describe('Testing App', () => {
  it('should render correctly', () => {
    useSelector.mockReturnValueOnce([mockVehicle])
    const { container } = render(<Details />);
    expect(container).toMatchSnapshot();
  });
  it('should render correctly empty', () => {
    useSelector.mockReturnValueOnce([])
    const { container } = render(<Details />);
    expect(container).toMatchSnapshot();
  });
});
