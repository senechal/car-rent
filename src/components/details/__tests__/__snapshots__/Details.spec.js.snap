// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App should render correctly 1`] = `
<div>
  <main
    class="rac-layout-content rac-details-content"
  >
    <div
      class="rac-details-navigation"
    >
      <a
        href="/"
      >
         
        <span
          aria-label="double-left"
          class="anticon anticon-double-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="double-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M272.9 512l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L186.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H532c6.7 0 10.4-7.7 6.3-12.9L272.9 512zm304 0l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L490.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H836c6.7 0 10.4-7.7 6.3-12.9L576.9 512z"
            />
          </svg>
        </span>
         Go Back
      </a>
    </div>
    <section
      class="rac-vehicle"
    >
      <div
        class="rac-vehicle-content"
      >
        <div
          class="rac-vehicle-img"
        >
          <a
            href="/details/HERTZ_1"
          >
            <img
              alt="Kia Rio or similar"
              src="https://ctimg-fleet.cartrawler.com/kia/rio/primary.png?auto=format&w=600"
            />
          </a>
        </div>
        <div
          class="rac-vehicle-info"
        >
          <div
            class="rac-vehicle-make-model"
          >
            <a
              href="/details/HERTZ_1"
            >
              <strong>
                Kia Rio
              </strong>
               or 
              similar
               (
              Available
              )
            </a>
          </div>
          <div
            class="rac-vehicle-features"
          >
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  seats.svg
                </svg>
              </span>
               
              5
               seats 
            </span>
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  doors.svg
                </svg>
              </span>
               
              2
               doors 
            </span>
          </div>
          <div
            class="rac-vehicle-features"
          >
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  bag.svg
                </svg>
              </span>
               
              2
               bags 
            </span>
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  ac.svg
                </svg>
              </span>
               
              Air Conditioning
               
            </span>
          </div>
          <div
            class="rac-vehicle-features"
          >
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  fuel.svg
                </svg>
              </span>
               
              Petrol
               
            </span>
            <span>
              <span
                class="anticon rac-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  automatic.svg
                </svg>
              </span>
               Automatic Transmission
            </span>
          </div>
        </div>
        <div
          class="rac-vehicle-pricing"
        >
          CA$731.47
        </div>
      </div>
      <div
        class="rac-vehicle-vendor"
      >
        <strong>
          Vendor: 
        </strong>
        <img
          alt="HERTZ"
          src="hertz_logo.gif"
        />
      </div>
    </section>
  </main>
</div>
`;

exports[`Testing App should render correctly empty 1`] = `
<div>
  &lt;Redirect to="/"&gt;
</div>
`;
